libs/ssh/setup_test.go

```diff
@@ -0,0 +1,416 @@
+package ssh
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/experimental/mocks"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateClusterAccess(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("valid cluster with single user mode", func(t *testing.T) {
```

Comment:
There is no shared code or setup between these tests.

Instead of defining `t.Run` inside a function, the more common approach is to use separate functions:
```
TestValidateClusterAccess_SingleUser
TestValidateClusterAccess_InvalidAccessMode
TestValidateClusterAccess_NotFound
```

It's more common to use `t.Run` for table tests.